runtime/common/EvolveResult.h

```diff
@@ -100,20 +100,23 @@ class evolve_result {
     return final_state;
   }
 
-  std::optional<std::vector<state>> get_intermediate_states() {
+  const std::optional<std::vector<state>> &get_intermediate_states() {
```

Comment:
The spec proposal is dropping these getters in favor of public std::optional members